components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -0,0 +1,992 @@
+     *
+     * @param request            CIBA Authentication Request.
+     * @param cibaAuthRequestDTO DTO that  captures validated parameters.
+     * @throws CibaAuthFailedException Ciba Authentication Failed Exception.
+     */
+    public void validateAuthRequestParameters(String request, CibaAuthRequestDTO cibaAuthRequestDTO)
+            throws CibaAuthFailedException {
+
+        try {
+
+            long currentTime = ZonedDateTime.now().toInstant().toEpochMilli();
+            long skewTime = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000;
+            SignedJWT signedJWT = SignedJWT.parse(request);
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+            JSONObject authRequestAsJSON = signedJWT.getJWTClaimsSet().toJSONObject();
+
+            if (isValidSignature(signedJWT).equals(false)) {
+                //Signature is invalid.
+                throw new CibaAuthFailedException(HttpServletResponse.SC_UNAUTHORIZED, ErrorCodes.UNAUTHORIZED_CLIENT,
+                        ErrorCodes.SubErrorCodes.INVALID_SIGNATURE);
+
+            }
+
+            // Validate audience of the Request.
+            validateAudience(claimsSet, cibaAuthRequestDTO);
+
+            // Validate  JWT-ID of the Request.
+            validateJWTID(claimsSet, cibaAuthRequestDTO);
+
+            // Validate the expiryTime of the Request.
+            validateExpiryTime(claimsSet, cibaAuthRequestDTO, currentTime, skewTime);
+
+            // Validate the issuedTime of the Request.
+            validateIssuedTime(claimsSet, cibaAuthRequestDTO, currentTime);
+
+            // Validate the NBF of the Request.
+            validateNBF(claimsSet, cibaAuthRequestDTO, currentTime, skewTime);
+
+            // Validate the scope of the Request.
+            validateScope(authRequestAsJSON, cibaAuthRequestDTO);
+
+            // Validate the client_notification_token of the Request.
+            validateClientNotificationToken(authRequestAsJSON, cibaAuthRequestDTO);
+
+            // Validate the client_notification_token of the Request.
+            validateACRValues(authRequestAsJSON, cibaAuthRequestDTO);
+
+            // Validate the binding_message of the Request.
+            validateBindingMessage(authRequestAsJSON, cibaAuthRequestDTO);
+
+            // Validate the transaction_context of the Request.
+            validateTransactionContext(authRequestAsJSON, cibaAuthRequestDTO);
+
+            // Validate the requested_expiry of the Request.
+            validateRequestedExpiry(authRequestAsJSON, cibaAuthRequestDTO);
+
+            if (log.isDebugEnabled()) {
+                log.debug(" CIBA Authentication Request made by client with clientID : " +
+                        cibaAuthRequestDTO.getAudience() + "is properly validated.");
+            }
+
+        } catch (ParseException e) {
+            throw new CibaAuthFailedException(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
```

Comment:
let's change the Exception to 'CibaAuthFailureException'

Reply:
Changed.